### PR TITLE
Expanded SKY130

### PR DESCRIPTION
**Pull request by ThomasPluck**
This expands the SKY130 PDK to include all `fd_pr` primitives as described in #103.

A couple of changes to base HDL21 `primitives.py` are needed including:

- Add `model` parameter to `BipolarParams`
- Add `ShieldedCapacitor` primitive for Perp/Parr VPP primitives
- Add `W/L` to `PhysicalCapacitorParams` to account for generic/varactor variables

Have also expanded the repertoire of tests to include all new devices with compilation and netlisting.

Need to add more comprehensive simulation tests to build out `test_site.py` to ensure all components are working as expected, but I figured this would be a good first start.

`primitives.py` in `sky130` are me toying with the idea of introducing helper functions to make it even easier to create devices with `sky130` import

### CODE REVIEW

**Comment by ThomasPluck** [comment]
- Think this is now ready to go to review, default sizing for each of the components is cobbled together from device documentation + a bit of digging through xschem and slack channels for hidden lore.
- Built out "sanity tests" for all components just to be sure that each device is producing a netlist that agrees with what is known to work with ngspice. 
- Gave each device category its own Sky130 brand paramclass, fixed what I had to in Primitives to get appropriate netlisting and easy compilation semantics
- Might add ease-of-use primitives functions in a future commit/push.

File: hdl21/primitives.py [hunk 1]
```diff
@@ -400,6 +405,25 @@ class PhysicalCapacitorParams:
     aliases=["Cap3", "PhyCap3", "CapPhy3", "PhyCapacitor3"],
 )
 
+# For the 4 terminal components, we have
```
Comment (dan-fritchman) [inline hdl21/primitives.py:408]: I don't think we should add this one as a generic primitive.
I don't see how it'll clearly generalize to other technologies.  
Recommend we keep it in the PDK / ExternalModules.

File: hdl21/primitives.py [hunk 2]
```diff
@@ -531,7 +558,7 @@ class SineVoltageSourceParams:
     phase = Param(dtype=Optional[Scalar], default=None, desc="Phase at td (degrees)")
 
 
-_add(
+SineVoltageSource = _add(
```
Comment (dan-fritchman) [inline hdl21/primitives.py:561]: These assignments are helping with IDE references, right? 

Assuming so, a good thing to have. 
Some history: that `_add` method got "added" when one or more of them got outta sync with its `name` field, and that confused export/ import steps. 
With this as-is, `_add` is setting a module-level `SineVoltageSource` attribute, this assignment is over-riding it, and even if we, say, spelled this assignment `SignVoltageSource` (i.e. a typo), both should be available.

File: hdl21/primitives.py [hunk 3]
```diff
@@ -640,6 +667,8 @@ class BipolarParams:
     tp = Param(
         dtype=BipolarType, desc="Bipolar Type (NPN/ PNP)", default=BipolarType.NPN
     )
+    model = Param(dtype=Optional[str], desc="Model (Name)", default=None)
```
Comment (dan-fritchman) [inline hdl21/primitives.py:670]: Good additions. Noting these generic BJTs have not gotten much use to date, and could use some TLC!

File: hdl21/primitives.py [hunk 4]
```diff
@@ -686,6 +715,7 @@ def Pnp(arg: Any = Default, **kwargs) -> Primitive:
 class DiodeParams:
     w = Param(dtype=Optional[Scalar], desc="Width in resolution units", default=None)
     l = Param(dtype=Optional[Scalar], desc="Length in resolution units", default=None)
+    a = Param(dtype=Optional[Scalar], desc="Area in resolution units squared", default=None)
```
Comment (dan-fritchman) [inline hdl21/primitives.py:718]: Do we really need/ want this as an independent parameter? 
Is the fact that there are `w` and `l` parameters not implicative that the shape is a rectangle, and `a == w * l`?

**Comment by ThomasPluck** [comment]
Okay - I think this addresses the criticisms and works as intended, let me know what you think!

**Comment by dan-fritchman** [comment]
Awesome stuff!

**Pull request merged**